Source/Scene/MetadataTableProperty.js

```diff
@@ -0,0 +1,610 @@
+import Check from "../Core/Check.js";
+import ComponentDatatype from "../Core/ComponentDatatype.js";
+import defaultValue from "../Core/defaultValue.js";
+import defined from "../Core/defined.js";
+import FeatureDetection from "../Core/FeatureDetection.js";
+import getStringFromTypedArray from "../Core/getStringFromTypedArray.js";
+import oneTimeWarning from "../Core/oneTimeWarning.js";
+import MetadataType from "./MetadataType.js";
+
+/**
+ * A binary property in a metadata table.
+ *
+ * @param {Object} options Object with the following properties:
+ * @param {Number} options.count The number of entities in the table.
+ * @param {Object} options.property The property JSON object.
+ * @param {MetadataClassProperty} options.classProperty The class property.
+ * @param {Object} options.bufferViews An object mapping bufferView IDs to Uint8Array objects.
+ *
+ * @alias MetadataTableProperty
+ * @constructor
+ *
+ * @private
+ */
+function MetadataTableProperty(options) {
+  options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+  var count = options.count;
+  var property = options.property;
+  var classProperty = options.classProperty;
+  var bufferViews = options.bufferViews;
+
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.number.greaterThan("options.count", count, 0);
+  Check.typeOf.object("options.property", property);
+  Check.typeOf.object("options.classProperty", classProperty);
+  Check.typeOf.object("options.bufferViews", bufferViews);
+  //>>includeEnd('debug');
+
+  var i;
+
+  var isArray = classProperty.type === MetadataType.ARRAY;
+  var isVariableSizeArray = isArray && !defined(classProperty.componentCount);
+
+  var valueType = classProperty.valueType;
+  var enumType = classProperty.enumType;
+
+  var hasStrings = valueType === MetadataType.STRING;
+  var hasBooleans = valueType === MetadataType.BOOLEAN;
+
+  var offsetType = defaultValue(
+    MetadataType[property.offsetType],
+    MetadataType.UINT32
+  );
+
+  var arrayOffsets;
+  if (isVariableSizeArray) {
+    arrayOffsets = new BufferView(
+      bufferViews[property.arrayOffsetBufferView],
+      offsetType,
+      count + 1
+    );
+  }
+
+  var componentCount;
+  if (isVariableSizeArray) {
+    componentCount = 0;
+    for (i = 0; i < count; ++i) {
+      componentCount += arrayOffsets.get(i + 1) - arrayOffsets.get(i);
+    }
```

Comment:
This can be computed by subtracting `arrayOffsets(count) - arrayOffsets(0)`, as the middle terms cancel out in this loop.